internal/provider/resource_cloudflare_record.go

```diff
@@ -198,15 +198,14 @@ func resourceCloudflareRecordRead(ctx context.Context, d *schema.ResourceData, m
 		return diag.FromErr(err)
 	}
 
-	data, dataOk := d.GetOk("data")
```

Comment:
this is the crux of the bug; it would look in `d.Get` for the value which on Import would never exist. this should have always pulled from the API anyway to ensure the remote was the source of truth, not the local files.